File: app/src/main/java/org/amoustakos/boilerplate/util/ErrorUtil.java
package org.amoustakos.boilerplate.util;


public final class ErrorUtil {


    /**
     * Checks if the {@link Throwable} is a connection error.
     */
    public static boolean isConnectionError(Throwable t){
        return t instanceof java.net.ConnectException
                || t instanceof java.net.NoRouteToHostException
                || t instanceof java.net.PortUnreachableException
                || t instanceof java.net.ProtocolException
                || t instanceof java.net.SocketException
                || t instanceof java.net.SocketTimeoutException
                || t instanceof java.net.UnknownHostException
                || t instanceof java.net.UnknownServiceException;
    }





}

File: app/src/main/java/org/amoustakos/boilerplate/util/ErrorUtil.kt
package org.amoustakos.boilerplate.util


object ErrorUtil {


	/**
	 * Checks if the [Throwable] is a connection error.
	 */
	fun isConnectionError(t: Throwable): Boolean {
		return t is java.net.ConnectException
				|| t is java.net.NoRouteToHostException
				|| t is java.net.PortUnreachableException
				|| t is java.net.ProtocolException
				|| t is java.net.SocketException
				|| t is java.net.SocketTimeoutException
				|| t is java.net.UnknownHostException
				|| t is java.net.UnknownServiceException
	}


}
